### ISSUE TITLE
Strange issue on GKE

### ISSUE THREAD

**Issue by samstride** [kpacha/krakend-k8s#1]
Hi,

So I followed the guide to deploy the gateway, tweaked it to my needs and everything works well on local k8s via Docker Desktop.

These are the 2 tweaks I used:

- Use port 5000 for deployment
- Have both the deployment and service definitions in a single yaml file.

Now when I deploy the same thing to GKE on GCP, I get the following error:

```sh
ERROR parsing the configuration file: '/etc/krakend/krakend.json': 1 error(s) decoding:

* cannot parse 'port' as int: strconv.ParseInt: parsing "tcp://10.56.4.60:5000": invalid syntax
```

I checked the syntax of the JSON as follows:

```sh
krakend check -c krakend.json -d

Hosts: []
Extra (2):
  github: map[com/devopsfaith/krakend-ratelimit/juju/router:map[clientmaxrate:10 strategy:ip]]
  github_com/devopsfaith/krakend-gologging: map[level:WARNING prefix:[KRAKEND] stdout:true syslog:false]
Endpoints (2):
        REDACTED
Syntax OK!

```

If I split the `deployment` and `service` definitions into 2 different yaml's and deploy the `deployment` first and then the `service`, it works without any issues.

If I update the krakend.json file to add a new route, and deploy it again, I run into the same issue. I always have to have NO `service` definition for a `deployment` to work. Any ideas why this happens on GKE but not local k8s?

Thanks.

**Comment by samstride** [comment]
On GKE there is a collision with env variable `PORT`.

Resolved in v1.2.0 by adding

```yaml
 - name: KRAKEND_PORT
   value: "8080" // <-- change port value if needed
```

**Comment by LePau** [comment]
Here is my solution in the deployment manifest

```
spec:
  template:
    spec:
       enableServiceLinks: false
```

This prevents kubernetes from creating variables like KRAKEND_PORT
